ocaml/test/test_platformdata.ml

```diff
@@ -18,19 +18,24 @@ open Test_highlevel
 
 module SanityCheck = Generic.Make(struct
 	module Io = struct
-		type input_t = ((string * string) list * bool)
-		type output_t = (string * string) list
+		type input_t = ((string * string) list * bool * int64 * int64 * bool)
+		type output_t = (exn, (string * string) list) Either.t
 
 		let string_of_input_t input =
-			Printf.sprintf "(platformdata = %s, filter_out_unknowns = %b)"
-				(fst input |> Test_common.string_of_string_map)
-				(snd input)
+			let (platformdata, filter, vcpu_max, vcpu_startup, hvm) = input in
```

Comment:
You can do away with this line by deconstructing the tuple in the function definition, i.e.

```
let string_of_input_t (platformdata, filter, vcpu_max, vcpu_startup, hvm) = 
  ...
```